pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -210,7 +213,19 @@ def qfunc_transformed(*args, **kwargs):
             for op in transformed_tape.circuit:
                 qml.apply(op)
 
-            return transformed_tape._qfunc_output  # pylint:disable=protected-access
+            mps = transformed_tape.measurements
+
+            if not mps:
+                return qfunc_output
+
+            if isinstance(qfunc_output, qml.measurements.MeasurementProcess):
+                return tuple(mps) if len(mps) > 1 else mps[0]
+
+            if isinstance(qfunc_output, (tuple, list)):
+                return type(qfunc_output)(mps)
+
+            interface = qml.math.get_interface(qfunc_output)
+            return qml.math.asarray(mps, like=interface)
```

Comment:
this is the only notable change here imo. what's happening:
- if the transformed tape has no measurements, return the original qfunc_output. the `pattern_matching_optimization` was one such example needing this. I originally didn't check for it, but it made those tests fail, so I added it and they passed
- if it originally returned a single measurement, do the same but with the new measurements. If the transformed tape has more than one measurement, return it as a tuple since that is the default type for a sequence of tapes in PL
- if it originally returned a tuple (default) or a list, return the new measurements as the same type
- otherwise it must have originally returned some interface-specific array, so return with the same type